crypto/asn1/d2i_pr.c

```diff
@@ -51,6 +52,14 @@ d2i_PrivateKey_decoder(int keytype, EVP_PKEY **a, const unsigned char **pp,
     p8info = d2i_PKCS8_PRIV_KEY_INFO(NULL, pp, len);
     ERR_pop_to_mark();
     if (p8info != NULL) {
+        int64_t v;
+
+        /* ascertain version is 0 or 1 as per RFC5958 */
+        if (!ASN1_INTEGER_get_int64(&v, p8info->version)
+            || (v != 0 && v != 1)) {
+            ERR_raise(ERR_LIB_ASN1, ASN1_R_ASN1_PARSE_ERROR);
+            goto err;
```

Comment:
This should fix the CI issue:

```suggestion
            ERR_raise(ERR_LIB_ASN1, ASN1_R_ASN1_PARSE_ERROR);
            PKCS8_PRIV_KEY_INFO_free(p8info);
            return NULL;
```

Reply:
Embarassing. Thanks @t8m. Pointers helpful for how to set up a locally working memory test environment would be welcome, though: The test failed on master even in a clean ubuntu docker image :-(